staging/src/k8s.io/kubectl/pkg/cmd/auth/whoami.go

```diff
@@ -142,45 +152,74 @@ func NewCmdWhoAmI(restClientGetter genericclioptions.RESTClientGetter, streams g
 var (
 	notEnabledErr = fmt.Errorf(
 		"the selfsubjectreviews API is not enabled in the cluster\n" +
-			"enable APISelfSubjectReview feature gate and authentication.k8s.io/v1alpha1 API")
+			"enable APISelfSubjectReview feature gate and authentication.k8s.io/v1alpha1 or authentication.k8s.io/v1beta1 API")
 
 	forbiddenErr = fmt.Errorf(
 		"the selfsubjectreviews API is not enabled in the cluster or you do not have permission to call it")
 )
 
 // Run prints all user attributes.
 func (o WhoAmIOptions) Run() error {
-	sar := &authenticationv1alpha1.SelfSubjectReview{}
-	response, err := o.authClient.SelfSubjectReviews().Create(context.TODO(), sar, metav1.CreateOptions{})
+	var (
+		res runtime.Object
+		err error
+	)
+
+	res, err = o.authV1beta1Client.
+		SelfSubjectReviews().
+		Create(context.TODO(), &authenticationv1beta1.SelfSubjectReview{}, metav1.CreateOptions{})
 	if err != nil {
-		switch {
-		case errors.IsForbidden(err):
-			return forbiddenErr
-		case errors.IsNotFound(err):
-			return notEnabledErr
-		default:
-			return err
+		if errors.IsNotFound(err) {
+			// Fallback to Alpha API if Beta is not enabled
+			res, err = o.authV1alpha1Client.
+				SelfSubjectReviews().
+				Create(context.TODO(), &authenticationv1alpha1.SelfSubjectReview{}, metav1.CreateOptions{})
+			if err != nil {
+				return checkEnabledOrForbidden(err)
+			}
+		} else {
+			return checkEnabledOrForbidden(err)
 		}
 	}
```

Comment:
can drop `checkEnabledOrForbidden` and simplify to:

```go
if err != nil && errors.IsNotFound(err) {
	// Fallback to Alpha API if Beta is not enabled
	res, err = o.authV1alpha1Client.
		SelfSubjectReviews().
		Create(context.TODO(), &authenticationv1alpha1.SelfSubjectReview{}, metav1.CreateOptions{})
}
if err != nil {
	switch {
	case errors.IsForbidden(err):
		return forbiddenErr
	case errors.IsNotFound(err):
		return notEnabledErr
	default:
		return err
	}
}
```

Reply:
Great, I will go with your suggestion. Thanks.